harness/determined/searcher/search_runner.py

```diff
@@ -0,0 +1,112 @@
+                    continue
+                logging.warning(json.dumps([e.to_json() for e in events.searcherEvents]))
+                for e in events.searcherEvents:
+                    if e.initialOperations:
+                        logging.info("initial operations")
+                        operations = self.search_method.initial_operations()
+                    elif e.trialCreated:
+                        logging.info(f"trialCreated({e.trialCreated.requestId})")
+                        operations = self.search_method.on_trial_created(
+                            uuid.UUID(e.trialCreated.requestId)
+                        )
+                    elif e.trialClosed:
+                        logging.info(f"trialClosed({e.trialClosed.requestId})")
+                        operations = self.search_method.on_trial_closed(
+                            uuid.UUID(e.trialClosed.requestId)
+                        )
+                    elif e.trialExitedEarly:
+                        logging.info(
+                            f"trialExitedEarly({e.trialExitedEarly.requestId},"
+                            f" {e.trialExitedEarly.exitedReason})"
+                        )
+                        if e.trialExitedEarly.exitedReason is None:
+                            raise RuntimeError(
+                                "trialExitedEarly event is invalid without exitedReason"
+                            )
+                        operations = self.search_method.on_trial_exited_early(
+                            uuid.UUID(e.trialExitedEarly.requestId),
+                            exited_reason=ExitedReason._from_bindings(
+                                e.trialExitedEarly.exitedReason
+                            ),
+                        )
+                    elif e.validationCompleted:
+                        logging.info(
+                            f"validationCompleted({e.validationCompleted.requestId},"
+                            f" {e.validationCompleted.metric})"
+                        )
+                        if e.validationCompleted.metric is None:
+                            raise RuntimeError(
+                                "validationCompleted event is invalid without a metric"
+                            )
+                        operations = self.search_method.on_validation_completed(
+                            e.validationCompleted.metric
+                        )
+                    elif e.experimentInactive:
+                        logging.info(
+                            f"experiment {experiment_id} is inactive"
+                            f" state={e.experimentInactive.experimentState}"
+                        )
+                        experiment_is_active = False
+                        break
+                    else:
+                        raise RuntimeError(f"Unsupported event {e}")
+                    bindings.post_PostSearcherOperations(
```

Comment:
It doesn't matter a whole lot, but I was angling for the operations from multiple events to be batched together, and it sounded like that was fine, but it's not happening here.